### PR TITLE
feat(datasource/docker): Support OCI image manifests

### LINKED ISSUE #10046: Support reading OCI Docker/Container manifests
**What Renovate type, platform and version are you using?**

**Renovate type**

- [X] Hosted app
- [ ] Self hosted

**Renovate platform**

- [ ] Azure DevOps (dev.azure.com)
- [ ] Azure DevOps Server (self-hosted)
- [ ] Bitbucket Cloud (bitbucket.org)
- [ ] Bitbucket Server (self-hosted)
- [ ] Gitea (gitea.com, self-hosted)
- [X] GitHub (github.com, GitHub Enterprise)
- [ ] GitLab (gitlab.com, self-hosted)

**Describe the bug**

If a OCI container manifests exists in a registry Renovate fails to check it erroring with a 404 error

**Relevant debug logs**

<details><summary>Click me to see logs</summary>

```
{
  "err": {
    "name": "HTTPError",
    "timings": {
      "start": 1621424626953,
      "socket": 1621424626953,
      "lookup": 1621424626956,
      "connect": 1621424626967,
      "secureConnect": 1621424626978,
      "upload": 1621424626978,
      "response": 1621424627548,
      "end": 1621424627549,
      "phases": {
        "wait": 0,
        "dns": 3,
        "tcp": 11,
        "tls": 11,
        "request": 0,
        "firstByte": 570,
        "download": 1,
        "total": 596
      }
    },
    "message": "Response code 404 (Not Found)",
    "stack": "HTTPError: Response code 404 (Not Found)\n    at Request.<anonymous> (/home/ubuntu/renovateapp/node_modules/renovate/node_modules/got/dist/source/as-promise/index.js:117:42)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)",
    "options": {
      "headers": {
        "user-agent": "Renovate Bot (GitHub App 2740)",
        "authorization": "***********",
        "accept": "application/vnd.docker.distribution.manifest.list.v2+json, application/vnd.docker.distribution.manifest.v2+json",
        "accept-encoding": "gzip, deflate, br"
      },
      "url": "https://ghcr.io/v2/anthr76/tumbleweed-dnf-toolbox/manifests/latest",
      "username": "",
      "password": "",
      "method": "GET",
      "http2": false
    },
    "response": {
      "statusCode": 404,
      "statusMessage": "Not Found",
      "body": "{\"errors\":[{\"code\":\"MANIFEST_UNKNOWN\",\"message\":\"OCI manifest found, but Accept header does not support OCI manifests\"}]}\n",
      "headers": {
        "content-type": "application/json",
        "docker-distribution-api-version": "registry/2.0",
        "date": "Wed, 19 May 2021 11:43:47 GMT",
        "content-length": "122",
        "x-github-request-id": "86CC:8278:33934:8986B:60A4F9F2",
        "connection": "close"
      },
      "httpVersion": "1.1"
    }
  },
  "registry": "https://ghcr.io",
  "dockerRepository": "anthr76/tumbleweed-dnf-toolbox",
  "tag": "latest"
}
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/main/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [X] I have provided a minimal reproduction repository
- [X] I don't have time for that, but it happens in a public repository I have linked to : https://github.com/anthr76/tooling , https://github.com/anthr76/teamredminer-container
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

In the linked repos it's best to check the workflows, specifically that these images exists in GHCR as a OCI compliant manifest. https://github.com/opencontainers/image-spec/blob/master/manifest.md

### PR DESCRIPTION
**Pull request by bpfoster**
## Changes

Support for the OCI image manifests (`application/vnd.oci.image.manifest.v1+json`) added to the docker datasource.  The structure is identical to the docker v2 manifest which makes things easier.  The only hiccup is the spec does not require top-level `mediaType`, and indeed GitHub Container Registry does not provide it.

## Context

Closes #10046

## Documentation (please check one with an [x])

- [ ] I have updated the documentation, or
- [x] No documentation update is required

## How I've tested my work (please tick one)

I have verified these changes via:

- [ ] Code inspection only, or
- [ ] Newly added/modified unit tests, or
- [ ] No unit tests but ran on a real repository, or
- [x] Both unit tests + ran on a real repository

Reproduction repository: https://github.com/bpfoster/renovate-10046-reproduction

### CODE REVIEW

File: lib/datasource/docker/__snapshots__/index.spec.ts.snap [hunk 1]
```diff
@@ -14,6 +21,22 @@ Object {
 }
 `;
 
+exports[`datasource/docker/index getReleases supports OCI manifests with media type 1`] = `
+Object {
+  "registryUrl": "https://registry.company.com",
+  "releases": Array [],
```
Comment (viceice) [inline lib/datasource/docker/__snapshots__/index.spec.ts.snap:27]: If it supports this, there should be some releases.

File: lib/datasource/docker/__snapshots__/index.spec.ts.snap [hunk 2]
```diff
@@ -14,6 +21,22 @@ Object {
...
+  "sourceUrl": "https://github.com/renovatebot/renovate",
+}
+`;
+
+exports[`datasource/docker/index getReleases supports OCI manifests without media type 1`] = `
+Object {
+  "registryUrl": "https://registry.company.com",
+  "releases": Array [],
```
Comment (viceice) [inline lib/datasource/docker/__snapshots__/index.spec.ts.snap:35]: Here too

File: lib/datasource/docker/types.ts [hunk 3]
```diff
@@ -9,6 +9,7 @@ export enum MediaType {
   manifestV1 = 'application/vnd.docker.distribution.manifest.v1+json',
   manifestV2 = 'application/vnd.docker.distribution.manifest.v2+json',
   manifestListV2 = 'application/vnd.docker.distribution.manifest.list.v2+json',
+  ociManifestV1 = 'application/vnd.oci.image.manifest.v1+json',
```
Comment (JamieMagee) [inline lib/datasource/docker/types.ts:12]: Should we also add support for `application/vnd.oci.image.index.v1+json`[^1]? The OCI equivalent of docker's manifest list

[^1]: https://github.com/opencontainers/image-spec/blob/main/image-index.md

Comment (viceice) [reply lib/datasource/docker/types.ts:12]: Sounds good

**Review by JamieMagee — DISMISSED** [review]

**Comment by JamieMagee** [comment]
LGTM. Just waiting for #14490 to go in first.

**Comment by bpfoster** [comment]
#14490 has been merged.  Any chance this can move forward now?

**Review by JamieMagee — APPROVED** [review]

**Pull request merged**